src/Tree/Document/MongoDB/Repository/AbstractTreeRepository.php

```diff
@@ -21,6 +21,9 @@
 use Gedmo\Tree\RepositoryUtilsInterface;
 use Gedmo\Tree\TreeListener;
 
+/**
+ * @phpstan-extends DocumentRepository<object>
+ */
```

Comment:
Based on https://github.com/doctrine-extensions/DoctrineExtensions/pull/2556 and https://github.com/doctrine-extensions/DoctrineExtensions/pull/2543 we should maybe make it generic

Reply:
As this PR is focused on resolving current findings and not in adding more extension points, I guess we can let that responsibility on those PRs.
WDYT?

Reply:
👍 just wondering if, as a user, static analysis issues would appear, I guess not because there are not in `tests`.